packages/app-store/routing-forms/lib/processRoute.tsx

```diff
@@ -41,41 +40,30 @@ export function processRoute({
     // After above flat map, all routes are non router routes.
     .concat([fallbackRoute]) as z.infer<typeof zodNonRouterRoute>[];
 
-  routesWithFallbackInEnd.some((route) => {
+  for (const route of routesWithFallbackInEnd) {
```

Comment:
Converted from .some to for loop because  otherwise TypeScript can't determine if chosenRoute was ever set. 

It allows me to be able to remove this type assertion
![image](https://github.com/user-attachments/assets/b350ad9d-c907-4b80-b702-da57f07f1a0b)